core/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -327,6 +328,36 @@ public void pause() {
         this.state.updateAndGet(s -> s.isRunning() ? State.PAUSED : s);
     }
 
+    /**
+     * Indicates whether this tracking processor can constructs a new {@code Thread} to process
+     * another segment of events.
+     *
+     * @return {@code true} if tracking processor can constructs a new {@code Thread}, {@code false} otherwise
+     */
+    public boolean hasAvailableThreads() {
+        return maxThreadCount > activeProcessorThreads();
+    }
+
+    /**
+     * Blacklist a segment identifier only if it is currently claimed by this instance.
+     * This causes the segment to be released and not longer to be claimed until it's removed from the black list.
+     *
+     * @param segmentId the id of the segment to be blacklisted
+     */
+    public void blacklistSegment(int segmentId) {
+        if (activeSegments.containsKey(segmentId)) {
+            segmentsToRelease.add(segmentId);
+        }
+    }
+
+    private void releaseSegment(int segmentId){
+        activeSegments.remove(segmentId);
+        if (segmentsToRelease.contains(segmentId)){
+            doSleepFor(DEFAULT_BACKOFF_TIME_MILLIS * 2);
```

Comment:
Instead of doing a sleep here, could we attach a timeout to the blacklisted segments? That would allow this thread to immediately become a Launcher thread. Perhaps there is another segment that this thread can grab.

The blacklist would probably need to be a map, where the value is the timestamp at which it may be released. So if a segment is in the map, and the current time is before the value of the entry, it's blacklisted. Otherwise, remove the entry.